mobile/src/actions/contactsBackup.js

```diff
@@ -0,0 +1,158 @@
+          multiple: false
+        }
+      )
+    }
+  })
+}
+
+// versions 0.3.5 of the app and prior only requested permissions for io.cozy.files ; this function makes checks if we have permissions for io.cozy.contact and requests them if not
+export const requestCozyAuthorization = () => (dispatch, getState) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const tokenScope = getState().mobile.settings.tokenScope || ''
+      if (tokenScope.indexOf(DOCTYPE_CONTACTS) >= 0) {
+        resolve(true)
+      } else {
+        await dispatch(registerDevice())
+        resolve(true)
+      }
+    } catch (e) {
+      logException(e)
+      resolve(false)
+    }
+  })
+}
+
+const loadDeviceContacts = () => {
+  return new Promise((resolve, reject) => {
+    navigator.contacts.find(['*'], resolve, reject)
+  })
+}
+
+const getCozyContacts = async (ids = []) => {
+  let response
+  try {
+    response = await cozy.client.fetchJSON('GET', '/data/io.cozy.contacts/_all_docs?include_docs=true', {keys: ids})
+  } catch (e) {
+    // exceptions might happen if the database is not created yet, which is ok
+    return []
+  }
+  return response.rows.map(row => row.doc)
+}
+
+const cordovaContactToCozy = (contact) => {
+  let cozyContact = {}
+  let baseContact = {
+    displayName: '',
+    name: {},
+    emails: [],
+    addresses: [],
+    phoneNumbers: [],
+    ...contact
+  }
+
+  cozyContact.fullname = baseContact.displayName
+
+  cozyContact.name = {
+    familyName: baseContact.name.familyName,
+    givenName: baseContact.name.givenName,
+    additionalName: baseContact.name.middleName,
+    namePrefix: baseContact.name.honorificPrefix,
+    nameSuffix: baseContact.name.honorificSuffix
+  }
+
+  cozyContact.email = []
+  baseContact.emails.forEach(email => {
+    cozyContact.email.push({
+      address: email.value,
+      type: email.type,
+      label: null,
+      primary: !!email.pref
+    })
+  })
+
+  cozyContact.address = []
+  baseContact.addresses.forEach(address => {
+    cozyContact.address.push({
+      street: address.streetAddress,
+      pobox: null,
+      city: address.locality,
+      region: address.region,
+      postcode: address.postalCode,
+      country: address.country,
+      type: address.type,
+      primary: !!address.pref,
+      formattedAddress: address.formatted
+    })
+  })
+
+  cozyContact.phone = []
+  baseContact.phoneNumbers.forEach(phone => {
+    cozyContact.phone.push({
+      number: phone.value,
+      type: phone.type,
+      label: null,
+      primary: !!phone.pref
+    })
+  })
+
+  return cozyContact
+}
+
+export const backupContacts = () => async (dispatch) => {
+  let deviceContacts
+
+  try {
+    deviceContacts = await loadDeviceContacts()
+  } catch (e) {
+    // the authorization has been revoked
+    dispatch(setBackupContacts(false))
+    return
+  }
+  // keep only contacts with an email address
```

Comment:
why don't you create a named function to get such contacts?

```js
function filterContactsWithEmailAddress (contacts) {
   return contacts.filter()
}
```

Then, first you don't need a comment line to express the intention. Second you don't add out-of-scope responsibility to the main function `backupContacts`.